src/verifier/nativeVerifier.ts

```diff
@@ -0,0 +1,62 @@
+import { VerifierOptions } from './types';
+import { getVerifierLib } from '../ffi/verifier';
+import logger from '../logger';
+import { argMapping } from './arguments';
+import { argumentMapper } from '../ffi/argumentMapper';
+
+const VERIFICATION_SUCCESSFUL = 0;
+const VERIFICATION_FAILED = 1;
+// 2 - null string passed
+// 3 - method panicked
+const INVALID_ARGUMENTS = 4;
+
+export const verify = (opts: VerifierOptions): Promise<string> => {
+  const verifierLib = getVerifierLib();
+  // Todo: probably separate out the sections of this logic into separate promises
+  return new Promise<string>((resolve, reject) => {
+    // Todo: Does this need to be a specific log level?
```

Comment:
This accepts the name of an environment variable that can be set.

e.g. if you did `LOG_LEVEL=trace npm run ...` it would print rust level trace. Or put another way, the environment variable is what actually sets the log level.

Reply:
Right, I'll make it set and pass down the env variable.